src/App.js

```diff
@@ -2,13 +2,37 @@ import Navigation from "./components/Navigation";
 import AppHeader from "./components/Header";
 import HomePage from "./pages/homePage";
 import { Routes, Route } from "react-router-dom";
+import Warenkorb from "./pages/Warenkorb";
+import History from "./pages/History";
+import data from "./components/Data";
+import { useState } from "react";
+import { search } from "fast-fuzzy";
 
 function App() {
+  const { dataItems } = data;
+  const [filteredData, setFilteredData] = useState([]);
+
+  function inputValue(value) {
+    let filtered = search(value, dataItems, {
```

Comment:
hi Angela, i think it is necessary to use let in this case, cause every time you enter something in input field you will get another result (array) depending on your request. What about search() -it is the "fast-fuzzy-search". it looks through your array and filters it.

Reply:
Hey,
the function `inputValue()` gets called every time you make a keypress in your input field. Since variables, that are declared inside functions, are tossed away after the function call is over, the filtered variable will not be over-written, but newly created every time the function is triggered on a new key press. Just try to exchange the `let` with a `const` and test whether it is still working. ;D